intervaltree/interval.py

```diff
@@ -53,6 +53,31 @@ def overlaps(self, begin, end=None):
         except:
             return self.contains_point(begin)
 
+    def overlap_size(self, begin, end=None):
+        """
+        Return the overlap size between two intervals or a point
+        :param begin: beginning point of the range, or the point, or an Interval
+        :param end: end point of the range. Optional if not testing ranges.
+        :return: Return the overlap size, None if not overlap is found
+        :rtype: int
+        """
+        overlaps = self.overlaps(begin, end)
+        if not overlaps:
+            return None
+
+        if end is not None:
+            # case end is given
+            i0 = max(self.begin, begin)
+            i1 = min(self.end, end)
+            return i1 - i0
+        try:
+            # assume this is
+            i0 = max(self.begin, begin.begin)
+            i1 = min(self.end, begin.end)
+            return i1 - i0
+        except:
+            return 1  # case the begin is point
```

Comment:
Since this will catch *all* exceptions, we might get this return value even if we didn't get a point, but instead got something crazy and obviously wrong, say, a tuple or a dict.

I think it would be better to let the exception be thrown. It doesn't in general make sense to ask for the overlap size with a point, since we know the answer is (technically) always 0. Although most users will be using ints, this package is designed to also work with reals and dates, and any other types that can be sorted, added and subtracted. A return of 1 isn't always useful or correct.

Reply:
:+1: